tests/v2/validation/registrysetup/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+
+	"github.com/rancher/lasso/pkg/log"
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+	"github.com/rancher/rancher/tests/framework/pkg/environmentflag"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/v2/validation/registries"
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	testSession := session.NewSession()
+
```

Comment:
Then we do this in setup 
```
environmentFlags := environmentflag.NewEnvironmentFlags()
 	environmentflag.LoadEnvironmentFlags(environmentflag.ConfigurationFileKey, environmentFlags)
 	useRegistries := environmentFlags.GetValue(environmentflag.UseExistingRegistries)
```

to either create the registries or not.

Reply:
I've done the refactor cause a recommendation of taking all the environment setup outside the test suite. 

Do we want this back ?

Reply:
When I recommended that I was only referring to the HA setup, since that required a lot post setup steps that other future setups could use. The other registries I was fine with it being in the setup as long as there was flag to denote if they have been setup or not, sorry I should have been more clear. But I don't mind it too much would also like to hear @caliskanugur opinion on it. 

This is the last thing, and it's good to go.

Reply:
So actually I asked here too: https://github.com/rancher/rancher/pull/40930#discussion_r1142697297, in my opinion, I can see a solid advantage of creating the private registry in the test. That way Jenkins would have: 

1. Create HA with corral
2. Run any validation tests
3. Cleanup

I think since this is gonna be the first Jenkins with HA corral it has an important role, and by eliminating the private registry above, this Jenkinsfile can be reusable for Airgap, maybe Proxy and Local cluster upgrade too.

Reply:
Simply stated, the auth and no auth private registry corral with the flag should be done in the SetupSuite. One the rancher HA (registry) setup should be done in a separate bin

Reply:
Agreed